guides/source/testing.md

```diff
@@ -8,72 +8,90 @@ This guide covers built-in mechanisms in Rails for testing your application.
+adheres to the desired functionality even after some major code refactoring.
 
-Rails tests can also simulate browser requests and thus you can test your application's response without having to test it through your browser.
+Rails tests can also simulate browser requests and thus you can test your
+application's response without having to view the output through your browser.
 
 Introduction to Testing
 -----------------------
 
-Testing support was woven into the Rails fabric from the beginning. It wasn't an "oh! let's bolt on support for running tests because they're new and cool" epiphany.
+Testing was woven into the Rails fabric from the very beginning, and is central
+to the development process right from the creation of a new application.
 
-### Rails Sets up for Testing from the Word Go
+### Test Setup
 
-Rails creates a `test` directory for you as soon as you create a Rails project using `rails new` _application_name_. If you list the contents of this directory then you shall see:
+Rails creates a `test` directory for you as soon as you create a Rails project
+using `rails new` _application_name_. If you list the contents of this directory
+then you shall see:
 
 ```bash
 $ ls -F test
 application_system_test_case.rb  controllers/                     helpers/                         mailers/                         system/
 channels/                        fixtures/                        integration/                     models/                          test_helper.rb
 ```
 
-The `helpers`, `mailers`, and `models` directories are meant to hold tests for view helpers, mailers, and models, respectively. The `channels` directory is meant to hold tests for Action Cable connection and channels. The `controllers` directory is meant to hold tests for controllers, routes, and views. The `integration` directory is meant to hold tests for interactions between controllers.
+### Test Directories
+
+The `helpers`, `mailers`, and `models` directories are where tests for view
+helpers, mailers, and models respectively should be stored. The `channels`
+directory is where tests for Action Cable connection and channels should be
+stored. The `controllers` directory is where tests for controllers, routes, and
+views should be stored. The `integration` directory is where tests for
+interactions between controllers should be stored.
 
 The system test directory holds system tests, which are used for full browser
-testing of your application. System tests allow you to test your application
-the way your users experience it and help you test your JavaScript as well.
-System tests inherit from Capybara and perform in browser tests for your
-application.
+testing of your application. System tests allow you to test your application the
+way your users experience it and help you test your JavaScript as well. System
+tests inherit from Capybara and perform in-browser tests for your application.
```

Comment:
@Ridhwana having read back through this, I think it's best to leave as is, as this is the `Test Directories` section, so mainly focussing on navigating around where everything that we do have should live, rather than contrasting different approaches.